psydac/linalg/kron.py

```diff
@@ -499,14 +510,30 @@ def _allocate_temps(self):
         else:
             temp2 = np.empty((self._tempsize,), dtype=self._dtype)
         return temp1, temp2
+
+    @property
+    def domain(self):
+        return self._domain
     
     @property
-    def space(self):
-        """
-        Returns the space associated to this solver (i.e. where the information
-        about the cartesian distribution is taken from).
-        """
-        return self._space
+    def codomain(self):
+        return self._codomain
+    
+    @property
+    def dtype(self):
+        return None
+    
+    def toarray(self):
+        raise NotImplementedError('toarray() is not defined for KroneckerLinearSolvers.')
+    
+    def tosparse(self):
+        raise NotImplementedError('tosparse() is not defined for KroneckerLinearSolvers.')
+
+    def transpose(self, conjugate=False):
+        raise NotImplementedError('transpose() is not defined for KroneckerLinearSolvers.')
```

Comment:
In principle the transposed operator can be built because transposition is distributive over the Kronecker product. For this to work we would need the `LinearSolver` objects to provide a `transpose` method.

Reply:
Yes, and I think you should add the `transpose` method to the `LinearSolver` base class. Subclasses are free to raise a `NotImplementedError`.